pkg/rehearse/jobs_test.go

```diff
@@ -347,7 +302,8 @@ func TestExecuteJobsErrors(t *testing.T) {
 				return false, nil, nil
 			})
 
-			executor := NewExecutor(tc.jobs, testPrNumber, testRepoPath, testRefs, true, testLoggers, fakeclient)
+			rehearsals := ConfigureRehearsalJobs(tc.jobs, testRepoPath, testPrNumber, testLoggers, true)
+			executor := NewExecutor(rehearsals, testPrNumber, testRepoPath, testRefs, false, testLoggers, fakeclient)
```

Comment:
I'm not a fan of how these tests are actually silently changed (`dryRun` param). I know what you want to achieve here and the change actually improves the coverage, but it would deserve a separate commit with explanation, not to be buried with a refactor/unrelated feature commit.

Reply:
I agree with your point. I will exclude the `dryRun` param change from this PR. This change should be covered in #87